### PR TITLE
Feature/expose agent group in the scan

### PR DESCRIPTION
**Pull request by oussamaessaji**

### CODE REVIEW

**Review by ErebusZ — DISMISSED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -1043,6 +1043,25 @@ def scan(mocker: plugin.MockerFixture, db_engine_path: str) -> models.Scan:
         return scan
 
 
+@pytest.fixture
+def scan_with_agent_group(
+    mocker: plugin.MockerFixture, db_engine_path: str, agent_group: models.AgentGroup
```
Comment (deadly-panda) [inline tests/conftest.py:1048]: isn't this missing the `clean_db` fixture?

File: tests/conftest.py [hunk 2]
```diff
@@ -1043,6 +1043,25 @@ def scan(mocker: plugin.MockerFixture, db_engine_path: str) -> models.Scan:
...
+    """Create dummy scan with agent group."""
+    mocker.patch.object(models, "ENGINE_URL", db_engine_path)
+    with models.Database() as session:
+        scan = models.Scan(
+            title="Scan with agent group",
+            asset="Any",
+            progress=models.ScanProgress.DONE,
+            created_time=datetime.datetime.now(),
+            agent_group_id=agent_group.id,
+        )
```
Comment (elyousfi5) [inline tests/conftest.py:1059]: use create instead

File: src/ostorlab/serve_app/oxo.py [hunk 3]
```diff
@@ -690,6 +690,8 @@ def mutate(
 
                 with models.Database() as session:
                     created_scan.agent_group_id = scan.agent_group_id
+                    session.add(created_scan)
```
Comment (3asm) [inline src/ostorlab/serve_app/oxo.py:693]: Why?

**Review by 3asm — APPROVED** [review]

File: src/ostorlab/serve_app/oxo.py (hunk 3 above)
Comment (oussamaessaji) [reply src/ostorlab/serve_app/oxo.py:693]: The scan was never updated with the agent_group_id

**Review by elyousfi5 — APPROVED** [review]

File: src/ostorlab/serve_app/types.py [hunk 4]
```diff
@@ -334,6 +334,174 @@ class Meta:
...
+            )
+            return AgentsType(agents=agents)
+
+
+class AgentGroupsType(graphene.ObjectType):
+    agent_groups = graphene.List(AgentGroupType, required=True)
+    page_info = graphene.Field(common.PageInfo, required=False)
+
+
+class OxoAssetInputType(graphene.InputObjectType):
```
Comment (3asm) [inline src/ostorlab/serve_app/types.py:466]: Some are called Oxo, some are not where is the logic?

**Review by 3asm — CHANGES_REQUESTED** [review]

**Comment by 3asm** [comment]
Please resolve conflict.

File: src/ostorlab/serve_app/types.py [hunk 5]
```diff
@@ -341,6 +341,231 @@ class Meta:
...
+    """SQLAlchemy object type for an agent."""
+
+    args = graphene.Field(OxoAgentArgumentsType, required=True)
+
+    class Meta:
+        """Meta class for the agent object type."""
+
+        model = models.Agent
+        only_fields = (
+            "id",
```
Comment (3asm) [inline src/ostorlab/serve_app/types.py:391]: Why do we need id?

File: src/ostorlab/serve_app/types.py [hunk 6]
```diff
@@ -341,6 +341,231 @@ class Meta:
...
+    """Graphene object type for a list of agent arguments."""
+
+    value = common.Bytes(required=False)
+
+    class Meta:
+        """Meta class for the agent arguments object type."""
+
+        model = models.AgentArgument
+        only_fields = (
+            "id",
```
Comment (3asm) [inline src/ostorlab/serve_app/types.py:354]: Why do we need id?

File: src/ostorlab/serve_app/types.py [hunk 7]
```diff
@@ -341,6 +341,231 @@ class Meta:
...
+    )
+    asset_types = graphene.List(graphene.String)
+
+    class Meta:
+        """Meta class for the agent group object type."""
+
+        model = models.AgentGroup
+
+        only_fields = (
+            "id",
```
Comment (3asm) [inline src/ostorlab/serve_app/types.py:442]: id?

File: src/ostorlab/serve_app/types.py [hunk 8]
```diff
@@ -369,6 +595,7 @@ class Meta:
         only_fields = (
             "id",
             "title",
+            "agent_group_id",
```
Comment (3asm) [inline src/ostorlab/serve_app/types.py:598]: Why?

File: tests/serve_app/oxo_test.py [hunk 9]
```diff
@@ -2863,7 +2863,9 @@ def testOxoSchemaReOxoSchemas_whenMutations_schemasShouldBeSimilar() -> None:
             assert arg_type == oxo_mutation_args[arg_name]
 
 
-@pytest.mark.skip(reason="Schema not complete on RE_OXO.")
+@pytest.mark.skip(
+    reason="Asset type needs to be aligned (https://github.com/Ostorlab/oxo/pull/713)."
+)
```
Comment (3asm) [inline tests/serve_app/oxo_test.py:2868]: Why is not aligned?

File: tests/serve_app/oxo_test.py [hunk 10]
```diff
@@ -3092,3 +3094,33 @@ def testPublishAgentGroup_withoutNameAndAgentArgs_shouldPersistAgentGroup(
...
+def testQueryScan_always_shouldReturnScanWithAgentGroup(
+    authenticated_flask_client: testing.FlaskClient, scan_with_agent_group: models.Scan
+) -> None:
+    """Test query for scan with agent group."""
+
+    query = """
+        query scan($scanId: Int) {
+  scan(scanId:$scanId) {
+    id
+    agentGroup{
```
Comment (3asm) [inline tests/serve_app/oxo_test.py:3108]: agent, args, values where are those to ensure all is working?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/serve_app/types.py [hunk 11]
```diff
@@ -341,6 +341,231 @@ class Meta:
...
+class OxoAgentGroupCreateInputType(graphene.InputObjectType):
+    """Input object type for creating an agent group."""
+
+    name = graphene.String(required=False)
+    description = graphene.String(required=True)
+    agents = graphene.List(OxoAgentGroupAgentCreateInputType, required=True)
+    asset_types = graphene.List(graphene.String, required=False, default_value=[])
+
+
+class OxoAgentScanInputType(graphene.InputObjectType):
```
Comment (ohachimOs) [inline src/ostorlab/serve_app/types.py:563]: No docs for small classes?

File: src/ostorlab/serve_app/types.py (hunk 7 above)
Comment (elyousfi5) [reply src/ostorlab/serve_app/types.py:480]: the id is removed from agent_arg and arg, but for agent group i think we should keep it, we might for instance run a scan with the same agent group.

File: src/ostorlab/serve_app/types.py (hunk 11 above)
Comment (elyousfi5) [reply src/ostorlab/serve_app/types.py:606]: addressed

File: tests/serve_app/oxo_test.py (hunk 10 above)
Comment (elyousfi5) [reply tests/serve_app/oxo_test.py]: applied

**Pull request merged**